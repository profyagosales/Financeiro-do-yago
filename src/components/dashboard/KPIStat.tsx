import type { ReactNode } from 'react';

import { KPI_TONE_STYLES, type KpiTone } from './kpiTones.ts';

import { MotionCard } from '@/components/ui/MotionCard';
import { cn } from '@/lib/utils';

export type KPIStatProps = {
  label: string;
  value: ReactNode; // j√° formatado ou componente AnimatedNumber
  icon?: ReactNode;
  hint?: string;
  className?: string;
  tone?: KpiTone;
  loading?: boolean;
  asChild?: boolean; // permite passar um MotionCard custom via Slot pattern no futuro
};

export function KPIStat({ label, value, icon, hint, className, tone='slate', loading, asChild }: KPIStatProps) {
  const Wrapper: any = asChild ? 'div' : MotionCard;
  return (
    <Wrapper className={cn('flex items-center gap-3 p-4 lg:col-span-3', className)}>
      {icon && (
        <div className={cn('p-2 rounded-full shrink-0', KPI_TONE_STYLES[tone])} aria-hidden>
          {icon}
        </div>
      )}
      <div className="flex flex-col min-w-0">
        <span className="text-sm text-fg-muted truncate">{label}</span>
        <span className={cn('font-medium text-base sm:text-lg tabular-nums', loading && 'opacity-50')}>
          {value}
        </span>
        {hint && <span className="sr-only">{hint}</span>}
      </div>
  </Wrapper>
  );
}
