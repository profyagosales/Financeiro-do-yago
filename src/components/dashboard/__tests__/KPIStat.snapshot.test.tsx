import { render } from '@testing-library/react';
import { describe, expect, it, vi } from 'vitest';

import { KPIStat } from '../KPIStat';
vi.mock('@/components/ui/MotionCard', () => ({
  MotionCard: ({ children, className }: any) => <div data-mock="MotionCard" className={className}>{children}</div>
}));

describe('KPIStat snapshot', () => {
  it('bÃ¡sico', () => {
    const { container } = render(<KPIStat label="Saldo" value="R$ 10" />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
