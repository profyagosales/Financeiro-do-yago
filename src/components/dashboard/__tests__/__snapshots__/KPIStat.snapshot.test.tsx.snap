// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`KPIStat snapshot > b√°sico 1`] = `
<div
  class="flex items-center gap-3 p-4 lg:col-span-3"
  data-mock="MotionCard"
>
  <div
    class="flex flex-col min-w-0"
  >
    <span
      class="text-sm text-fg-muted truncate"
    >
      Saldo
    </span>
    <span
      class="font-medium text-base sm:text-lg tabular-nums"
    >
      R$ 10
    </span>
  </div>
</div>
`;
